src/test/java/org/stellar/sdk/TransactionTest.java

```diff
@@ -260,4 +287,164 @@ public void testConstructorWithSorobanData() throws IOException {
+                                .contract(new Address(contractId).toSCAddress())
+                                .key(Scv.toLedgerKeyContractInstance())
+                                .durability(ContractDataDurability.PERSISTENT)
+                                .bodyType(ContractEntryBodyType.DATA_ENTRY)
+                                .build())
+                        .build()))
+            .build();
+    RestoreFootprintOperation operation = RestoreFootprintOperation.builder().build();
+    Transaction transaction =
+        new Transaction(
+            AccountConverter.enableMuxed(),
+            account.getAccountId(),
+            Transaction.MIN_BASE_FEE,
+            account.getIncrementedSequenceNumber(),
+            new org.stellar.sdk.Operation[] {operation},
+            null,
+            new TransactionPreconditions(
+                null, null, BigInteger.ZERO, 0, new ArrayList<SignerKey>(), null),
+            sorobanData,
+            Network.TESTNET);
+    assertTrue(transaction.isSorobanTransaction());
+  }
+
+  @Test
+  public void testIsSorobanTransactionMultiOperations() {
+    KeyPair source =
+        KeyPair.fromSecretSeed("SCH27VUZZ6UAKB67BDNF6FA42YMBMQCBKXWGMFD5TZ6S5ZZCZFLRXKHS");
+
+    Account account = new Account(source.getAccountId(), 2908908335136768L);
+
+    String contractId = "CDYUOBUVMZPWIU4GB4XNBAYL6NIHIMYLZFNEXOCDIO33MBJMNPSFBYKU";
+    String functionName = "hello";
+    List<SCVal> params = Collections.singletonList(Scv.toSymbol("Soroban"));
+    InvokeHostFunctionOperation operation =
+        InvokeHostFunctionOperation.invokeContractFunctionOperationBuilder(
+                contractId, functionName, params)
+            .build();
+
+    Transaction transaction =
+        new Transaction(
+            AccountConverter.enableMuxed(),
+            account.getAccountId(),
+            Transaction.MIN_BASE_FEE,
+            account.getIncrementedSequenceNumber(),
+            new org.stellar.sdk.Operation[] {operation, operation},
+            null,
+            new TransactionPreconditions(
+                null, null, BigInteger.ZERO, 0, new ArrayList<SignerKey>(), null),
+            null,
+            Network.TESTNET);
+    assertFalse(transaction.isSorobanTransaction());
+  }
+
+  @Test
+  public void testIsSorobanTransactionBumpSequenceOperation() {
+    KeyPair source =
+        KeyPair.fromSecretSeed("SCH27VUZZ6UAKB67BDNF6FA42YMBMQCBKXWGMFD5TZ6S5ZZCZFLRXKHS");
+
+    Account account = new Account(source.getAccountId(), 2908908335136768L);
+    BumpSequenceOperation operation = new BumpSequenceOperation.Builder(0L).build();
+
+    Transaction transaction =
+        new Transaction(
+            AccountConverter.enableMuxed(),
+            account.getAccountId(),
+            Transaction.MIN_BASE_FEE,
+            account.getIncrementedSequenceNumber(),
+            new org.stellar.sdk.Operation[] {operation, operation},
```

Comment:
i think only single op needed in tx correct? otherwise isSorobanTransaction=false, but for non-tested reason, i.e. it's more than one op, it won't validate op instance mismatch yet.

Reply:
This test is to verify that a transaction containing multiple operations is not a soroban transaction. You can take a look at https://github.com/stellar/java-stellar-sdk/pull/518/files#diff-0114ab460186855a151f487001dae40013372c8430be55eeebdb6e24bfc12705R448
```java
assertFalse(transaction.isSorobanTransaction());
```

Reply:
there is already another test for asserting multi-op - `testIsSorobanTransactionMultiOperations` outcome, I think this test was intending to exercise a non-multi op code path but an incorrect op type, which seemed to be the intent, just pass one `operation` and it should verify same false outcome.

Reply:
I'm sorry, I made a mistake. I have already corrected it. 🤦